src/ucp/proto/proto_init.c

```diff
@@ -326,17 +326,26 @@ void ucp_proto_init_memreg_time(const ucp_proto_common_init_params_t *params,
                                 ucs_linear_func_t *memreg_time,
                                 ucp_proto_perf_node_t **perf_node_p)
 {
-    ucp_context_h context = params->super.worker->context;
-    ucp_proto_perf_node_t *perf_node;
+    ucp_context_h context            = params->super.worker->context;
+    ucp_proto_perf_node_t *perf_node = NULL;
     const uct_md_attr_v2_t *md_attr;
     ucp_md_index_t md_index;
     const char *md_name;
 
     *memreg_time = UCS_LINEAR_FUNC_ZERO;
 
     if (reg_md_map == 0) {
-        *perf_node_p = NULL;
-        return;
+        goto out;
+    }
+
+    if (context->rcache != NULL) {
+        perf_node = ucp_proto_perf_node_new_data("rcache lookup", "");
+
+        *memreg_time = ucs_linear_func_make(50.0e-9, 0);
```

Comment:
As we discussed, this patch change the performance estimation through changing several `mem_reg` overheads to one `lookup`. But we already have `double uct_md_rcache_overhead(const uct_md_rcache_config_t *rcache_config)` internal interface which is supposed to be defined for the same purpose.

Now, this function is used to set `md->reg_cost` for IB MD. Is that a good idea to reuse this function in that patch?

If so, then rcache lookup seems MD independent from my point of view, but this function take `rcache_config` to take the `RCACHE_OVERHEAD` user control into account. Shouldn't we move this user control to MD independent place?

@yosefe @brminich @rakhmets  WDYT?

Reply:
eventually we'll remove uct rcache md config, but since there is no equivalent configuration in UCP today I think we better keep this PR simple, as-is, to resolve the performance issue and don't do any better UCP rcache config for now.